### ISSUE TITLE
Changing Start Date doesn't remove task from Actionable view

### ISSUE THREAD

**Issue by SqAtx** [getting-things-gnome/gtg#1039]
This is related to other problems that I've noticed/filed on the megabranch, but I couldn't find a duplicate here. Similar to https://github.com/diegogangl/gtg/issues/21 in that the DB has the correct data, but the UI hasn't been updated.

This is on commit [2ecc997](https://github.com/getting-things-gnome/gtg/commit/2ecc997d3fb3f6d102615dafa83bda765cebd420)

### Steps

* Go to the Actionable tab
* Select a task (single click)
* Click "Start Tomorrow"

or:

* Go to the Actionable tab
* Open a task (double click)
* Change the start date to a date in the future

### Expected

The task disappears from the Actionable view.

This is the behaviour in GTG 0.6.

### Actual

The task is still visible in the Actionable view. I can update the view and make the disappear by:
* switching to (for example) Open and back to Actionable
* marking an Actionable task as done

**Comment by SqAtx** [comment]
Interestingly, the Actionable tab is updated as expected when you mark a child task as done and it unblocks its parent. The child disappears from the view (because it's now done) and the parent appears (because it's now actionable).

Side note: the other way doesn't work, but I'll create another ticket for it when I can take the time to reproduce it properly. If you add a child to an Actionable task, the now parent is supposed to disappear and the new child to appear. I think in 0.7, the child doesn't appear unless I actually restart GTG. But I'll confirm that later in a different ticket.

**Comment by SqAtx** [comment]
This came back :( I'm now on f501dfbf6a618d0e0425ed25ae8dab37694e407e

From the Actionable view:
- Clicking on the checkbox to mark a task as done doesn't make the task disappear:
![image](https://github.com/user-attachments/assets/83ac90f0-5582-4942-a9e9-f760a077337d)
Going to Open and back to Actionable makes is disappear.

- Adding a child to a task doesn't make the parent disappear:
![image](https://github.com/user-attachments/assets/726ddb92-07b4-4c85-ba9d-1abad0980941)
![image](https://github.com/user-attachments/assets/6fde0397-70b6-4b31-903a-695560142c06)
Going to Open and back to Actionable again does the right thing:
![image](https://github.com/user-attachments/assets/e6e06e1c-e6d9-4f3f-9563-1026d99608aa)

- Clicking on a task and clicking Start Tomorrow doesn't make it disappear. Again, going to Open and back to Actionable works.

I ran a git bisect for the last problem. It indeed worked after 6ceec800ee3329f2cf683b09ecd43d94e850a7ac. It found commit a89f715bf9879e20bb6da1050e38671cf6347425. And indeed, reverting that comit on master fixes that problem.

While bisecting, I also noticed #1130

**Comment by diegogangl** [comment]
Looks like this isn't happening anymore in Master (`291d49cabb4f4b104e9c9a3eb1adcf7b8e021072`)
